src/patches/dense_patch_computation.cpp

```diff
@@ -242,23 +322,70 @@ InstructionApplicationResult try_apply_instruction_direct_followup(
 
         return {nullptr, {}};
     }
-    else if (const auto* bell_init = std::get_if<BellPairInit>(&instruction.operation)) 
+    else if (auto* bell_init = std::get_if<BellPairInit>(&instruction.operation)) 
     {
-        auto routing_region = router.find_routing_ancilla(slice, bell_init->loc1.target, bell_init->loc1.op, bell_init->loc2.target, bell_init->loc2.op);
-        if(!routing_region) {
-            return {std::make_unique<std::runtime_error>(lstk::cat("No valid route found for Bell pair creation: ",
-                bell_init->loc1.target, ":", PauliOperator_to_string(bell_init->loc1.op), ",",
-                bell_init->loc2.target, ":", PauliOperator_to_string(bell_init->loc2.op))), {}};
+
+        if (!bell_init->counter)
+        {
+            auto routing_region = router.find_routing_ancilla(slice, bell_init->loc1.target, bell_init->loc1.op, bell_init->loc2.target, bell_init->loc2.op);
+            if(!routing_region) 
+            {
+                return {std::make_unique<std::runtime_error>(lstk::cat("No valid route found for Bell pair creation: ",
+                    bell_init->loc1.target, ":", PauliOperator_to_string(bell_init->loc1.op), ",",
+                    bell_init->loc2.target, ":", PauliOperator_to_string(bell_init->loc2.op))), {}};
+            }
+            else if (routing_region->cells.size() < 2) 
+            {
+                return {std::make_unique<std::runtime_error>(lstk::cat("Shortest route cannot be used for Bell pair creation: ",
+                    bell_init->loc1.target, ":", PauliOperator_to_string(bell_init->loc1.op), ",",
+                    bell_init->loc2.target, ":", PauliOperator_to_string(bell_init->loc2.op))), {}};
+            }
+
+            // Construct LocalInstructions based on route
+            std::vector<LocalInstruction> local_instructions;
+            for (size_t i=0; i<routing_region->cells.size()-1; i=i+2)
+            {
+                // Push a BellPrepare instruction with PatchID's depending on the case
+                if (i==0) 
+                    local_instructions.push_back({BellPrepare{bell_init->side2, std::nullopt, routing_region->cells[i].cell, routing_region->cells[i+1].cell}});
+                else if (i == routing_region->cells.size() - 2)
+                    local_instructions.push_back({BellPrepare{std::nullopt, bell_init->side1, routing_region->cells[i].cell, routing_region->cells[i+1].cell}});
+                else 
+                    local_instructions.push_back({BellPrepare{std::nullopt, std::nullopt, routing_region->cells[i].cell, routing_region->cells[i+1].cell}});
+            }
+            for (size_t i=2; i<routing_region->cells.size()-1; i=i+2)
```

Comment:
Something interesting going on with the indices here. Might be a bit more intuitive to have this loop starting at `1` and changing the offsets below